samples/nrf_desktop/src/modules/usb_state.c

```diff
@@ -248,21 +249,20 @@ static void device_status(enum usb_dc_status_code cb_status, const u8_t *param)
 		if (state == USB_STATE_SUSPENDED) {
 			LOG_WRN("USB reset in suspended state, ignoring");
 		} else {
-			new_state = USB_STATE_IDLE;
+			new_state = USB_STATE_POWERED;
 		}
 		break;
 
 	case USB_DC_SUSPEND:
 		__ASSERT_NO_MSG(state != USB_STATE_DISCONNECTED);
+		before_suspend = state;
 		new_state = USB_STATE_SUSPENDED;
-		/* Not supported yet */
 		LOG_WRN("USB suspend");
 		break;
 
 	case USB_DC_RESUME:
 		__ASSERT_NO_MSG(state == USB_STATE_SUSPENDED);
-		new_state = USB_STATE_IDLE;
-		/* Not supported yet */
+		new_state = before_suspend;
```

Comment:
can resume be called without earlier suspend?

Reply:
Probably it should not happen, in addition we got assert for this.